lib/Dialect/AIEVec/Utils/Utils.cpp

```diff
@@ -25,6 +25,43 @@
 using namespace mlir;
 
 namespace xilinx::aievec {
+
+static int64_t getLowerBoundValue(Value idx) {
```

Comment:
Would it be safer to change the return type of this function and std::optional<int64_t>() for cases where it is unable to infer the lower bound? In that case the caller is responsible for handling this case.

Reply:
It's certainly cleaner, and ready for further improvement. Right now, we assume that accesses whose alignment can't be determined at compile-time, are aligned. In the future, we should at least consider if we want to treat all those cases as unaligned, and generate alignment code. It's a tricky trade-off.